src/shared/utilities/timeoutUtils.ts

```diff
@@ -118,3 +123,41 @@ export async function waitUntil<T>(
         await new Promise(r => setTimeout(r, opt.interval))
     }
 }
+
+/**
+ * Utility function to wrap a Timeout token around a Promise.
```

Comment:
This doesn't explain anything, it just repeats the function signature :) _Why_ does this function exist, what is its purpose? 

- Is it analogous to `waitUntil()`? If so...
    -  rename it to `waitUntilTimeout` or something like that
    - use the docstring of waitUntil() as a starting point...